frontend/apps/reader/modules/readersearch.lua

```diff
@@ -152,6 +152,7 @@ function ReaderSearch:onShowFulltextSearchInput()
         text = _("Regular expression (hold for help)"),
         checked = self.use_regex,
         parent = self.input_dialog,
+        max_width = self.input_dialog._input_widget.width,
```

Comment:
You probably want that also above in check_button_case ("Case sensitive" may become very long in some language translations :)
Also, dunno if we should remove some padding/margin/indentation from this to keep the whole truncated text centered - or if it's better to have it like on your screenshot takes a bit more space (without overflowing, may be by change) so we see more of the truncated text.

Reply:
> "Case sensitive" may become very long in some language translations

Do you think so? Can anybody give an example? Russian is one of the longest languages.
We have three dialogs with this checkbutton.



> a bit more space

I looked at the Open with... dialog, having a bit more space.

Reply:
> Do you think so? Can anybody give an example?

We don't need any example: you have and use a max_width in the other checkbox - just provide it also for the first checkbox, no reason not too.

Reply:
> max_width should probably be a requirement

It is now a requirement, so let's add it everywhere.